consensus/service/src/byzantine_ledger/mod.rs

```diff
@@ -94,15 +95,29 @@ impl ByzantineLedger {
         logger: Logger,
     ) -> Self {
         // TODO: this should be passed in as an argument.
-        let scp_node: Box<dyn ScpNode<TxHash>> = {
+        let scp_node: Box<dyn ScpNode<ConsensusValue>> = {
             let tx_manager_validate = tx_manager.clone();
             let tx_manager_combine = tx_manager.clone();
             let current_slot_index = ledger.num_blocks().unwrap();
             let node = Node::new(
                 node_id.clone(),
                 quorum_set,
-                Arc::new(move |tx_hash| tx_manager_validate.validate(tx_hash)),
-                Arc::new(move |tx_hashes| tx_manager_combine.combine(tx_hashes)),
+                Arc::new(move |scp_value| match scp_value {
+                    ConsensusValue::TxHash(tx_hash) => tx_manager_validate.validate(tx_hash),
+                }),
+                Arc::new(move |scp_values| {
+                    let mut tx_hashes = Vec::new();
+
+                    for value in scp_values {
+                        match value {
+                            ConsensusValue::TxHash(tx_hash) => tx_hashes.push(*tx_hash),
```

Comment:
This is an extra copy that was not previously present. It might be possible to remove it by maybe changing the combine function signature to receive an iterator instead of a slice, but I didn't want to make this PR any bigger than absolutely necessary.

Reply:
So, where is `scp_values` coming from?